### PR TITLE
Francisco branch pull request

### PR DESCRIPTION
**Pull request by franciscovalentecastro**
Hello Khuyen ! While reviewing your project submission I found a couple of bugs in your implementation which gave incorrect classifications in both the 'Artificial' dataset and the MNIST. The only big bug/mistake was a missing negative sign in the 'gradient' calculation (and in your report too), so that's why your 'gradient descent' was kinda maximizing instead of minimizing the function. 

After this change everything started to work correctly. You can review the changes made in this pull-request.

- Fixed error calculating gradient of binary cross entropy.
- Take only class 0 and 1 from MNIST dataset.
- Added confusion matrix to study classification.
- Added accuracy calculation.

### CODE REVIEW

**Comment by khuyentran1401** [comment]
Thank you for the feedback and the pull request!
